packages/api-electron/main.js

```diff
@@ -17,49 +19,64 @@ program
 let win;
 
 function createWindow() {
-  const appJson = loadConfig();
-  if (appJson) {
+  loadConfig().then(appJson => {
     ssfElectron(appJson, program.symphony, program.developer);
-  } else {
+  }).catch(err => {
+    consoleError(err.message);
     process.exit();
-  }
+  });
 }
 
 function loadConfig() {
-  // the config json is passed to the ssf-electron bin script
-  const configFile = program.config ? process.cwd() + '/' + program.config : null;
-  const launchUrl = program.url;
-  if (configFile) {
-    if (fs.existsSync(configFile)) {
-      const config = JSON.parse(fs.readFileSync(configFile, 'utf8'));
+  if (program.config) {
+    return readConfigFile()
+      .then(data => new Promise((resolve, reject) => {
+        const config = JSON.parse(data);
 
-      if (launchUrl) {
-        // Overridden by parameter
-        config.url = launchUrl;
-      }
+        if (program.url) {
+          // Overridden by parameter
+          config.url = program.url;
+        }
 
-      if (config.url) {
-        return config;
-      }
-    } else {
-      fileError(configFile, 'Config file does not exist');
-      return null;
-    }
+        if (config.url) {
+          resolve(config);
+        } else {
+          reject(new Error('You must specify an URL (--url) or a config file containing an url (--config)'));
+        }
+      }));
   } else {
-    if (launchUrl) {
-      return {
-        url: launchUrl
-      };
-    }
+    return new Promise((resolve, reject) => {
+      if (program.url) {
+        resolve({
+          url: program.url
+        });
+      } else {
+        reject(new Error('You must specify an URL (--url) or a config file containing an url (--config)'));
+      }
+    });
   }
-
-  consoleError('You must specify an URL (--url) or a config file containing an url (--config)');
-  return null;
 }
 
-function fileError(filename, err) {
-  consoleError('Error with file: ' + filename);
-  consoleError(err);
+function readConfigFile() {
+  const filePath = program.config;
+  if (filePath.toLowerCase().startsWith('http://') || filePath.toLowerCase().startsWith('https://')) {
+    return download(filePath);
+  } else {
+    return new Promise((resolve, reject) => {
```

Comment:
If you convert the fs.readFile API into a promise-based interface, can you remove the need to manually construct a Promise?